recipes/iceoryx/all/conanfile.py

```diff
@@ -0,0 +1,244 @@
+        return {
+            "iceoryx_posh::iceoryx_posh": "iceoryx::posh",
+            "iceoryx_posh::iceoryx_posh_roudi": "iceoryx::posh_roudi",
+            "iceoryx_binding_c::iceoryx_binding_c": "iceoryx::binding_c",
+            "iceoryx_utils::iceoryx_utils": "iceoryx::utils"
+        }
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def config_options(self):
+        if self.options.toml_config:
+            self.requires("cpptoml/0.1.1")
+        if self.settings.os == "Linux":
+            self.requires("acl/2.3.1")
+    
+    def validate(self):
+        os = self.settings.os
+        compiler = self.settings.compiler
+        version = tools.Version(self.settings.compiler.version)
+        if compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 14)
+        if os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration(
+                'Using Iceoryx on Windows currently just possible with "shared=False"')
+        if (compiler == "gcc" or compiler == "clang") and compiler.libcxx != "libstdc++11":
+            raise ConanInvalidConfiguration(
+                'Using Iceoryx with gcc or clang on Linux requires "compiler.libcxx=libstdc++11"')
+        if os == "Linux" and compiler == "gcc" and version <= "5":
+            raise ConanInvalidConfiguration(
+                "Using Iceoryx with gcc on Linux requires gcc 6 or higher.")
+        if os == "Linux" and compiler == "gcc" and version == "6":
+            self.output.warn(
+                "Iceoryx package is compiled with gcc 6, it is recommended to use 7 or higher")
+            self.output.warn(
+                "GCC 6 will built with warnings.")
+        if compiler == "Visual Studio" and version < "16":
+            raise ConanInvalidConfiguration(
+                "Iceoryx is just supported for Visual Studio compiler 16 and higher.")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["TOML_CONFIG"] = self.options.toml_config
+        self._cmake.configure()
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        tools.rmdir(self._pkg_share)
+        tools.rmdir(self._pkg_cmake)
+        if self.options.toml_config:
+            tools.rename(
+                os.path.join(self._pkg_etc, "roudi_config_example.toml"),
+                os.path.join(self._pkg_bin, "roudi_config_example.toml")
```

Comment:
```suggestion
                os.path.join(self._pkg_bin, "roudi_config_example.toml")
```

Is it really needed? The file name is example. If is really necessary, please, move to self.package_folder/res, which is the used for configuration files in the package.

Reply:
hmmm actually again a good question - i would like to keep it and will move to res folder because : i think iceoryx is mostly used for embedded area and the component "roudi" is a executable. It could be deployed with the "deploy" generator of conan. (Actually we also do this in projects - the -g "deploy" functionality is pretty cool). and here when deploying a default configuration file would be cool. - i think i simple rename it to roudi_config.toml.